homeassistant/components/asuswrt/router.py

```diff
@@ -299,16 +299,15 @@ async def update_devices(self) -> None:
             _LOGGER.warning("Reconnected to ASUS router %s", self.host)
 
         self._connected_devices = len(wrt_devices)
-        consider_home: int = self._options.get(
+        consider_home = self._options.get(
             CONF_CONSIDER_HOME, DEFAULT_CONSIDER_HOME.total_seconds()
         )
-        track_unknown: bool = self._options.get(
-            CONF_TRACK_UNKNOWN, DEFAULT_TRACK_UNKNOWN
-        )
+        track_unknown = self._options.get(CONF_TRACK_UNKNOWN, DEFAULT_TRACK_UNKNOWN)
 
         for device_mac, device in self._devices.items():
             dev_info = wrt_devices.pop(device_mac, None)
-            device.update(dev_info, consider_home)
+            if isinstance(consider_home, int):
```

Comment:
The previous solution was better. As `isinstance` checks are relatively expensive.

Even better would be to structure the code in a way we preserve upstream typing.

Reply:
I very much agree!
But what do you mean with previous version was better? You mean its better to have it untyped?